src/prefect/flows.py

```diff
@@ -158,6 +159,15 @@ def __init__(
             List[Callable[[schemas.core.Flow, FlowRun, State], None]]
         ] = None,
     ):
+        # Validate states is passed as list
+        states = [on_completion, on_failure, on_cancellation, on_crashed]
+        for state in states:
+            if state is not None and not isinstance(state, list):
+                raise TypeError(
+                    f"Expected list of callable for {state}; got"
+                    f" {type(state).__name__} instead."
+                )
```

Comment:
I'd call these `hook_types`/`hooks` instead of `states`/`state`. 

We probably want to support other iterables like tuples and sets here. Perhaps we can try to cast it to a list? e.g. `on_completion = list(on_completion)` with a try/except that captures the `TypeError` that is raised if a non-iterable type is given — we'd then raise it again with a nicer error?

This message you wrote also made me notice we could be checking if the items in the list are callables.

Reply:
Yeah sure, makes sense.